src/Vegetation/Roots.jl

```diff
@@ -0,0 +1,432 @@
+
+
+A struct for holding parameters of the Root Model. Eventually to be used with ClimaParameters.
+$(DocStringExtensions.FIELDS)
+"""
+struct RootsParameters{FT <: AbstractFloat}
+    "controls the shape and steepness of conductance vs. pressure curve, for roots: unitless"
+    a_root::FT
+    "controls the steepness of the relative conductance vs. pressure curve, for roots: inverse MPa"
+    b_root::FT
+    "controls the shape and steepness of relative conductance vs. pressure curve, for stems: unitless"
+    a_stem::FT
+    "controls the steepness of the conductance vs. pressure curve, for stems: inverse MPa"
+    b_stem::FT
+    "the physical size of the stem, in moles"
+    size_reservoir_stem_moles::FT
+    "the physical size of the leaves, in moles"
+    size_reservoir_leaf_moles::FT
+    "water conductance in roots (moles/s/MPa) when pressure is zero, a maximum"
+    K_max_root_moles::FT
+    "water conductance in stems (moles/s/MPa) when pressure is zero, a maximum"
+    K_max_stem_moles::FT
+end
+
+"""
+    RootsModel{FT, PS, D, RE, T, B} <: AbstractVegetationModel{FT}
+
+Defines, and constructs instances of, the RootsModel type, which is used
+for simulation flow of water to/from soil, along roots of different depths,
+along a stem, to a leaf, and ultimately being lost from the system by
+transpiration. 
+
+This model can be used in standalone mode by prescribing the transpiration rate
+and soil pressure at the root tips, or with a dynamic soil model using `ClimaLSM`.
+
+$(DocStringExtensions.FIELDS)
+"""
+struct RootsModel{FT, PS, D, RE, T} <: AbstractVegetationModel{FT}
+    "Parameters required by the root model"
+    param_set::PS
+    "The root model domain, of type `AbstractVegetationDomain`"
+    domain::D
+    "The root extraction model, of type `AbstractRootExtraction`"
+    root_extraction::RE
+    "The transpiration model, of type `AbstractTranspiration`"
+    transpiration::T
+    "The model name, which must be :vegetation"
+    model_name::Symbol
+end
+
+function RootsModel{FT}(;
+    param_set,
+    domain::AbstractVegetationDomain{FT},
+    root_extraction::AbstractRootExtraction{FT},
+    transpiration::AbstractTranspiration{FT},
+) where {FT}
+    args = (param_set, domain, root_extraction, transpiration)
+    return RootsModel{FT, typeof.(args)...}(args..., :vegetation)
+end
+
+"""
+    prognostic_vars(model::RootsModel)
+
+A function which returns the names of the prognostic 
+variables of the `RootsModel`.
+"""
+prognostic_vars(model::RootsModel) = (:rwc,)
+
+"""
+    function flow(
+        z_do::FT,
+        z_up::FT,
+        p_do::FT,
+        p_up::FT,
+        a::FT,
+        b::FT,
+        Kmax::FT,
+    ) where {FT}
+
+Computes the flow of water (moles/sec)  given the height and pressures
+at two points, `do` and `up`. Here, `a`, `b, and `Kmax` are parameters
+which parameterize the hydraulic conductance of the pathway along which
+the flow occurs.
+"""
+function flow(
```

Comment:
What are `do` and `up`? I'm also wondering if the name can be more descriptive - where is this flow occurring? Is it in a root or in any length of a plant stem/root?

Reply:
I believe "do" and "up" refer to downstream and upstream, but that the model does not require them to be ordered in this way. agreed that it can be clearer! This formula would hold in any length of a plant stem, root, or leaf, assuming steady state flow and constant hydraulic properties & geometry in that segment.

FWIW, this model has to be updated to apply to a bulk plant (or plants per unit ground area) - right now it is sort of for a single plant, but that wont generalize nicely to an LSM with very coarse resolution. but the hookup it demonstrates between soil & roots is still representative.